Classes/Controllers/ApplicationController.m

```diff
@@ -130,6 +130,30 @@ - (void) windowWillClose: sender
 	[firstResponder terminate: sender];
 }
 
+//Override the default behavior
+- (IBAction)openDocument:(id)sender {
+	NSOpenPanel* panel = [[NSOpenPanel alloc] init];
+	
+	[panel setCanChooseFiles:false];
+	[panel setCanChooseDirectories:true];
+	
+	[panel beginWithCompletionHandler:^(NSInteger result) {
+		if (result == NSFileHandlingPanelOKButton) {
+			PBRepositoryDocumentController* controller = [PBRepositoryDocumentController sharedDocumentController];
+			[controller openDocumentWithContentsOfURL:panel.URL display:true completionHandler:^(NSDocument * _Nullable document, BOOL documentWasAlreadyOpen, NSError * _Nullable error) {
+				if (!document) {
+					NSLog(@"Error opening repository \"%@\": %@", panel.URL.path, error);
+					[controller presentError:error];
+					[sender replyToOpenOrPrint:NSApplicationDelegateReplyFailure];
+				}
+				else {
+					[sender replyToOpenOrPrint:NSApplicationDelegateReplySuccess];
```

Comment:
I haven't tried removing them but I can confirm that messaging the application does work.

https://github.com/samsonjs/gitx/commit/38b324c044e980d0186fa1a9dac6d3f121c4d761